RHIDP-1959: Add information about Operator update channels (#283)
redhat-developer/red-hat-developers-documentation-rhdh: modules/admin/proc-install-operator.adoc

* You are logged in as an administrator on the {ocp-short} web console.
* You have configured the appropriate roles and permissions within your project to create an application. For more information, see the link:https://docs.openshift.com/container-platform/{ocp-version}/applications/index.html[Red Hat OpenShift documentation on Building applications].

[NOTE]
====

> For enhanced security, deploy the {product} Operator in a dedicated default namespace such as `rhdh-operator`. The cluster administrator can restrict other users' access to the operator resources through role bindings or cluster role bindings. You can choose to deploy the operator in the `openshift-operators` namespace instead, however, the {product} operator shares the namespace with other operators, and therefore any users who can create workloads in that namespace can get their privileges escalated from all Operators' service accounts.

Reviewer: ```suggestion
For enhanced security, deploy the {product} Operator in a dedicated default namespace such as `rhdh-operator`. The cluster administrator can restrict other users' access to the operator resources through role bindings or cluster role bindings. Should you choose to deploy the operator in the default `openshift-operators` namespace, be advised that doing so could allow any users who can create workloads in that namespace can get their privileges escalated from all Operators' service accounts. This is generally not recommended due to the security risk. 
```

Author: Similarly to a [previous PR we had](https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/253/files), if something isn't recommended, especially for security reasons, we should either say "do not do X" or just remove mentioning it in the docs. I am removing this line.

====

.Procedure

. In the *Administrator* perspective of the {ocp-short} web console, navigate to *Operators* -> *OperatorHub*.
. In the *Filter by keyword* box, enter {product-short} and click the *{product}* Operator card.
. On the *{product} Operator* page, click *Install*.

. On the *Install Operator* page, use the *Update channel* drop-down menu to choose an update channel:

> ** The *fast* channel provides minor version (y-stream) updates, for example, updating from version 1.0 to 1.1.

Reviewer: ```suggestion
** The *fast* channel provides minor version (y-stream) updates, for example, updating from version 1.0 to 1.1 to 1.2 and beyond.
```

Author: Updated and added a note per Slack thread with @rm3l

> ** The *fast-1.1* channel provides z-stream updates, for example, updating from version 1.1.1 to 1.1.2.

Reviewer: ```suggestion
** The *fast-1.2* channel provides z-stream updates, for example, updating from version 1.2.0 to 1.2.1.
```

. On the *Install Operator* page, choose the *Update approval* strategy for the Operator:
** If you choose the *Automatic* option, the Operator is updated without requiring manual confirmation.

> ** If you choose the *Manual* option, when a new update is released in the update channel, the update must be manually approved before installation can begin.

Reviewer: ```suggestion
** If you choose the *Manual* option, when a new update is released in the update channel, a notification will appear within the operator subscription, and the update must be manually approved my an administrator before installation can begin. If you are installing in an airgapped or proxied environment where it is required that you copy container images into your cluster manually, this is the recommended strategy to ensure seamless upgrades can occur once the airgap preparation has been done. 
```

Author: Updated in line with this comment

* You are logged in as an administrator on the {ocp-short} web console.
* You have configured the appropriate roles and permissions within your project to create an application. For more information, see the link:https://docs.openshift.com/container-platform/{ocp-version}/applications/index.html[Red Hat OpenShift documentation on Building applications].

[NOTE]
====
For enhanced security, deploy the {product} Operator in a dedicated default namespace such as `rhdh-operator`. The cluster administrator can restrict other users' access to the Operator resources through role bindings or cluster role bindings.
====

.Procedure

> . In the *Administrator* perspective of the {ocp-short} web console, navigate to *Operators* -> *OperatorHub*.

Reviewer 2: ```suggestion
. In the *Administrator* perspective of the {ocp-short} web console, navigate to *Operators > OperatorHub*.
```
IBM guidance is to use the greater than symbol. Looks like it's also supposed to be bold.
https://www.ibm.com/docs/en/ibm-style?topic=elements-menus-navigation#navigation-trees

. In the *Filter by keyword* box, enter {product-short} and click the *{product}* Operator card.
. On the *{product} Operator* page, click *Install*.

> . On the *Install Operator* page, use the *Update channel* drop-down menu to choose an update channel:

Reviewer 2: ```suggestion
. On the *Install Operator* page, use the *Update channel* drop-down menu to select the update channel that you want to use:
```
_click_ might also work here, but IBM advises against the word _choose_ in reference to UI actions. 
https://www.ibm.com/docs/en/ibm-style?topic=word-usage#c

** The *fast* channel provides y-stream (x.y) and z-stream (x.y.z) updates, for example, updating from version 1.1 to 1.2, or from 1.1.0 to 1.1.1.
+
[IMPORTANT]
====

> The `fast` channel includes all updates, any may introduce unexpected changes in your {product} deployment. Check the release notes for details about any potentially breaking changes.

Reviewer 2: ```suggestion
The `fast` channel includes all of the updates available for a particular version. Any update might introduce unexpected changes in your {product} deployment. Check the release notes for details about any potentially breaking changes.
```
You can take or leave the changes to the first sentence, but I do recommend breaking this into 2 sentences and uses "might" instead of "may".

> ====

Reviewer 2: is there a specific reason why we format _fast_ as bold in some places and as monospace text in other places?

Author: Bold when it's a UI element, backticks for general use because it's the literal name :)

** The *fast-1.1* channel only provides z-stream updates, for example, updating from version 1.1.1 to 1.1.2. If you want to update the {product} y-version in the future, for example, updating from 1.1 to 1.2, you must switch to the *fast* channel manually.

. On the *Install Operator* page, choose the *Update approval* strategy for the Operator:
** If you choose the *Automatic* option, the Operator is updated without requiring manual confirmation.

> ** If you choose the *Manual* option, when a new update is released in the update channel, a notification opens, and the update must be manually approved by an administrator before installation can begin.

Reviewer 2: ```suggestion
** If you choose the *Manual* option, a notification opens when a new update is released in the update channel, and the update must be manually approved by an administrator before installation can begin.
```

* You are logged in as an administrator on the {ocp-short} web console.
* You have configured the appropriate roles and permissions within your project to create an application. For more information, see the link:https://docs.openshift.com/container-platform/{ocp-version}/applications/index.html[Red Hat OpenShift documentation on Building applications].

[NOTE]
====
For enhanced security, deploy the {product} Operator in a dedicated default namespace such as `rhdh-operator`. The cluster administrator can restrict other users' access to the Operator resources through role bindings or cluster role bindings.
====

.Procedure

. In the *Administrator* perspective of the {ocp-short} web console, navigate to *Operators* -> *OperatorHub*.
. In the *Filter by keyword* box, enter {product-short} and click the *{product}* Operator card.
. On the *{product} Operator* page, click *Install*.

. On the *Install Operator* page, use the *Update channel* drop-down menu to choose an update channel:

> ** The *fast* channel provides y-stream (x.y) and z-stream (x.y.z) updates, for example, updating from version 1.0 to 1.1, or from 1.1.0 to 1.1.1.

Reviewer: As there was no operator in 1.0, this should be amended.

```suggestion
** The *fast* channel provides y-stream (x.y) and z-stream (x.y.z) updates, for example, updating from version 1.1 to 1.2, or from 1.1.0 to 1.1.1.
```

+
[IMPORTANT]
====

> The `fast` channel includes all updates, any may introduce breaking changes in your {product} deployment.

Reviewer: ```suggestion
The `fast` channel includes all updates, any may introduce unexpected changes in your {product} deployment.
```

====

> ** The *fast-1.1* channel only provides z-stream updates, for example, updating from version 1.1.1 to 1.1.2. If you want to update the {product} y-version in the future, for example, updating from 1.0 to 1.1, you must switch to the *fast* channel manually.

Reviewer: Again, there was no operator in 1.0.

```suggestion
** The *fast-1.1* channel only provides z-stream updates, for example, updating from version 1.1.1 to 1.1.2. If you want to update the {product} y-version in the future, for example, updating from 1.2 to 1.3, you must switch to the *fast* channel manually.
```

. On the *Install Operator* page, choose the *Update approval* strategy for the Operator:
** If you choose the *Automatic* option, the Operator is updated without requiring manual confirmation.
** If you choose the *Manual* option, when a new update is released in the update channel, a notification opens, and the update must be manually approved by an administrator before installation can begin.
+
[NOTE]
====

> If you are installing in an airgapped or proxied environment where it is required that you copy container images into your cluster manually, use this update strategy to ensure seamless upgrades can occur after the airgap preparation has been completed.

Reviewer: for context we could link to the section on airgapped operator install, eg. to https://access.redhat.com/documentation/en-us/red_hat_developer_hub/1.1/html-single/administration_guide_for_red_hat_developer_hub/index#proc-install-rhdh-airgapped-environment-ocp-operator_admin-rhdh

Author: I'm actually not sure we should mention this at all looking at the docs - this procedure is about using the OCP web console, whereas https://access.redhat.com/documentation/en-us/red_hat_developer_hub/1.1/html-single/administration_guide_for_red_hat_developer_hub/index#proc-install-rhdh-airgapped-environment-ocp-operator_admin-rhdh uses only the CLI, so I'm not sure how this would fit in. I would say maybe instead to remove this note so that we're not having scope creep on this PR, and if we need to talk about channels for airgapped installations, open a new Jira to update the docs for that?
